Error with cmake version 4.0.0

Using either nest 3.7 and nestml 7 or nest 3.8 and nestml 8, within a python 3.10 environment:

I tried to compile a nestml file with the python interface: 
```python
from os import makedirs
from os.path import dirname
from pynestml.frontend.pynestml_frontend import generate_target

build_dir = "nest_build"
module_name = "cerebmodule"
model_dir = dirname(__file__)
makedirs(build_dir)

generate_target(
    input_path=model_dir,
    target_platform="NEST",
    target_path=build_dir,
    module_name=module_name,
)
```

but with the most recent cmake version 4.0, I got the following  error:
```
WARNING:root:PyGSL is not available. The stiffness test will be skipped.
WARNING:root:Error when importing: No module named 'pygsl'

              -- N E S T --
  Copyright (C) 2004 The NEST Initiative

 Version: 3.8.0
 Built: Apr  4 2025 18:02:29

 This program is provided AS IS and comes with
 NO WARRANTY. See the file LICENSE for details.

 Problems or suggestions?
   Visit https://www.nest-simulator.org

 Type 'nest.help()' to find out more about NEST.


              -- N E S T --
  Copyright (C) 2004 The NEST Initiative

 Version: 3.8.0
 Built: Apr  4 2025 18:02:29

 This program is provided AS IS and comes with
 NO WARRANTY. See the file LICENSE for details.

 Problems or suggestions?
   Visit https://www.nest-simulator.org

 Type 'nest.help()' to find out more about NEST.

CMake Warning:
  Ignoring empty string ("") provided on the command line.


CMake Warning (dev) at CMakeLists.txt:93 (project):
  cmake_minimum_required() should be called prior to this top-level project()
  call.  Please see the cmake-commands(7) manual for usage documentation of
  both commands.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- The CXX compiler identification is GNU 11.4.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done

-------------------------------------------------------
cerebmodule Configuration Summary
-------------------------------------------------------

C++ compiler         : /usr/bin/c++
Build static libs    : OFF
C++ compiler flags   : 
NEST compiler flags  :  -std=c++17 -Wall -fopenmp  -O2 -fdiagnostics-color=auto
NEST include dirs    :  -I/home/drodarie/workspace/venv2/include/nest -I/usr/include -I/usr/include -I/usr/include -I/usr/lib/x86_64-linux-gnu/openmpi/include -I/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi
NEST libraries flags : -L/home/drodarie/workspace/venv2/lib/nest -lnest -lsli /usr/lib/x86_64-linux-gnu/libltdl.so /usr/lib/x86_64-linux-gnu/libreadline.so /usr/lib/x86_64-linux-gnu/libncurses.so /usr/lib/x86_64-linux-gnu/libgsl.so /usr/lib/x86_64-linux-gnu/libgslcblas.so   /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_cxx.so /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so /usr/lib/gcc/x86_64-linux-gnu/11/libgomp.so /usr/lib/x86_64-linux-gnu/libpthread.a

-------------------------------------------------------

You can now build and install 'cerebmodule' using
  make
  make install

The library file libcerebmodule.so will be installed to
  /home/drodarie/workspace/venv2/lib/nest
The module can be loaded into NEST using
  (cerebmodule) Install       (in SLI)
  nest.Install(cerebmodule)   (in PyNEST)

CMake Error in CMakeLists.txt:
  No cmake_minimum_required command is present.  A line of code such as

    cmake_minimum_required(VERSION 4.0)

  should be added at the top of the file.  The version specified may be lower
  if you wish to support older CMake versions for this project.  For more
  information run "cmake --help-policy CMP0000".


-- Configuring incomplete, errors occurred!
Traceback (most recent call last):
  File "/home/drodarie/workspace/venv2/lib/python3.10/site-packages/pynestml/codegeneration/nest_builder.py", line 171, in build
    subprocess.check_call(cmake_cmd, stderr=stderr, stdout=stdout, shell=shell,
  File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '-Dwith-nest=/home/drodarie/workspace/venv2/bin/nest-config', '', '.']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/drodarie/workspace/dbbs/cerebellum/cerebellum/nest_models/build_models.py", line 56, in <module>
    _build_nest_models()
  File "/home/drodarie/workspace/dbbs/cerebellum/cerebellum/nest_models/build_models.py", line 48, in _build_nest_models
    generate_target(
  File "/home/drodarie/workspace/venv2/lib/python3.10/site-packages/pynestml/frontend/pynestml_frontend.py", line 215, in generate_target
    return process()
  File "/home/drodarie/workspace/venv2/lib/python3.10/site-packages/pynestml/frontend/pynestml_frontend.py", line 518, in process
    _builder.build()
  File "/home/drodarie/workspace/venv2/lib/python3.10/site-packages/pynestml/codegeneration/nest_builder.py", line 181, in build
    raise GeneratedCodeBuildException(stages_exception[current_stage])
pynestml.exceptions.generated_code_build_exception.GeneratedCodeBuildException: Error occurred during cmake! More detailed error messages can be found in stderr.
```
The model compilation was working with cmake 3.22.1

--- a/pynestml/codegeneration/resources_nest/point_neuron/setup/CMakeLists.txt.jinja2
+++ b/pynestml/codegeneration/resources_nest/point_neuron/setup/CMakeLists.txt.jinja2
@@ -49,6 +49,8 @@
 # For more informations on how to extend and use your module see:
 #           https://nest.github.io/nest-simulator/extension_modules
 
+cmake_minimum_required( VERSION 3.19 )
+
 # 1) Name your module here, i.e. add later with -Dexternal-modules=${moduleName}:
 set( SHORT_NAME {{moduleName}} )
 

--- a/pynestml/codegeneration/resources_nest_compartmental/cm_neuron/setup/CMakeLists.txt.jinja2
+++ b/pynestml/codegeneration/resources_nest_compartmental/cm_neuron/setup/CMakeLists.txt.jinja2
@@ -49,6 +49,8 @@
 # For more informations on how to extend and use your module see:
 #           https://nest.github.io/nest-simulator/extension_modules
 
+cmake_minimum_required( VERSION 3.19 )
+
 # 1) Name your module here, i.e. add later with -Dexternal-modules=${moduleName}:
 set( SHORT_NAME {{moduleName}} )
 

